File: profile/src/main/java/com/artaeum/profile/client/UaaClient.java
package com.artaeum.profile.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@FeignClient(name = "uaa")
public interface UaaClient {

    @GetMapping("/uaa/users/id/{login}")
    Long getUserIdByLogin(@PathVariable String login);

    @GetMapping("/uaa/users/login/{id}")
    String getUserLoginById(@PathVariable Long id);
}

File: media/src/main/scala/com/artaeum/media/client/UaaClient.scala
package com.artaeum.media.client

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.{GetMapping, PathVariable}

@FeignClient(name = "uaa")
trait UaaClient {

  @GetMapping(Array("/uaa/users/id/{login}"))
  def getUserIdByLogin(@PathVariable login: String): Long

  @GetMapping(Array("/uaa/users/login/{id}"))
  def getUserLoginById(@PathVariable id: Long): String
}
